src/class/audio/audio_device.c

```diff
@@ -1079,6 +1081,52 @@ static inline bool audiod_fb_send(uint8_t rhport, audiod_function_t *audio)
 }
 #endif
 
+#if CFG_TUD_AUDIO_ENABLE_INTERRUPT_EP
+static bool set_int_ctr_number(audiod_function_t *audio)
+{
+
+  uint8_t const *p_desc = audio->p_desc;
+  // Get pointer at end
+  uint8_t const *p_desc_end = audio->p_desc + audio->desc_length - TUD_AUDIO_DESC_IAD_LEN;
+
+
+  // p_desc starts at required interface with alternate setting zero
+  while (p_desc < p_desc_end)
+  {
+    // Find correct interface
+    if (tu_desc_type(p_desc) == TUSB_DESC_INTERFACE && ((tusb_desc_interface_t const * )p_desc)->bInterfaceNumber == 0 && ((tusb_desc_interface_t const * )p_desc)->bAlternateSetting == 0)
```

Comment:
`bInterfaceNumber` could be other than 0, it's defined by `_intfnum` https://github.com/hathach/tinyusb/blob/28f49c088bb0c498d730d80943017172061cfd05/src/device/usbd.h#L442.

@PanRe could you take a look ? I havn't work on UAC for a moment 🤣

Reply:
Ah sorry, this is leftover from when I initially hacked it together for my specific project.  I'll generalise that soon.

Also, since there are several places where we sequentially search the descriptors for some specific descriptor, should I create a separate function for doing that, which takes a predicate as an argument?
It should reduce code-size a bit while not burning a significant amount of extra cycles as a result.
And the CPU overhead shouldn't matter much anyway since this is set-up stuff rather than "in the loop" stuff.

Reply:
> Also, since there are several places where we sequentially search the descriptors for some specific descriptor, should I create a separate function for doing that, which takes a predicate as an argument?

I think there is no need to do it now, later we can generalize them with a function like `tu_desc_search()` if needed.

Reply:
True, this number can be any interface number!

Reply:
You may also skip that check since you already work on the descriptor corresponding to the right interrupt EP. Do we need to remember which INT EP corresponds to which interface? For sending messages via INT EP?

Reply:
Done